chore(http): Remove range request example no longer returning 'accept-ranges: none' (#35852)
mdn/content: files/en-us/web/http/range_requests/index.md

{{HTTPSidebar}}

An HTTP {{HTTPHeader("Range")}} request asks the server to send parts of a resource back to a client.

> Range requests are useful for clients like media players that support random access, data tools that know they need only part of a large file, and download managers that let the user pause and resume a download.

Reviewer: nit to improve clarity and to remove "know"
```suggestion
Range requests are useful for various clients, including media players that support random access, data tools that require only part of a large file, and download managers that let users pause and resume a download.
```

## Checking if a server supports partial requests

> If an HTTP response includes the {{HTTPHeader("Accept-Ranges")}} header with any value other than `none`, the server supports range requests.

Author: Side note: some older browsers relied on `Accept-Ranges: none` response header to disable the 'pause' and 'resume' buttons in the download manager. Nowadays:

>If responses omit the `Accept-Ranges` header, it indicates the server doesn't support partial requests.

If responses omit the `Accept-Ranges` header, it indicates the server doesn't support partial requests.
If range requests are not supported, applications can adapt to this condition; for instance, download managers can disable pause buttons that relied on range requests to resume a download.

> To check if a server supports range requests, you can issue a {{HTTPMethod("HEAD")}} request so that you can inspect headers without requesting the resource in full.

Reviewer: ```suggestion
To check if a server supports range requests, you can issue a {{HTTPMethod("HEAD")}} request to inspect headers without requesting the resource in full.
```

The {{HTTPHeader("Content-Length")}} header is also helpful as it indicates the total size of the image if you were to make the same request using the `GET` method instead.

## Requesting a specific range from a server

If the server supports range requests, you can specify which part (or parts) of the document you want the server to return by including the {{HTTPHeader("Range")}} header in a HTTP request.

### Single part ranges

We can request a single range from a resource using curl for illustration.
The `-H` option appends a header line to the request, which in this case is the `Range` header requesting the first 1024 bytes.
The last option is `--output -` which will allow printing the binary output to the terminal:

```bash
curl https://i.imgur.com/z4d4kWk.jpg -i -H "Range: bytes=0-1023" --output -
```

The issued request looks like this:

```http
GET /z4d4kWk.jpg HTTP/2
Host: i.imgur.com
User-Agent: curl/8.7.1
Accept: */*
Range: bytes=0-1023
```

The server responds with a {{HTTPStatus("206", "206 Partial Content")}} status:

```http
HTTP/2 206
content-type: image/jpeg
content-length: 1024
content-range: bytes 0-1023/146515
…

(binary content)
```

> The {{HTTPHeader("Content-Length")}} header indicates the size of the requested range (and not the full size of the image).
> The {{HTTPHeader("Content-Range")}} response header indicates where in the full resource this partial message belongs.

Reviewer: ```suggestion
The {{HTTPHeader("Content-Length")}} header indicates the size of the requested range, not the full size of the image.
The {{HTTPHeader("Content-Range")}} response header indicates where this partial message belongs within the full resource.
```

curl http://www.example.com -i -H "Range: bytes=0-50, 100-150"
```

> The server responds with the {{HTTPStatus("206", "206 Partial Content")}} status as follows.

Reviewer: ```suggestion
The server responds with the {{HTTPStatus("206", "206 Partial Content")}} status as shown below.
```

The response contains a {{HTTPHeader("Content-Type")}} header, indicating that a multipart byterange follows.

> The boundary string (`3d6b6a416f9b5` in this case) separates each body part, which has its own `Content-Type` and `Content-Range` fields:

Reviewer: ```suggestion
The boundary string (`3d6b6a416f9b5` in this case) separates the body parts, each of which has its own `Content-Type` and `Content-Range` fields:
```
